code/modules/antagonists/changeling/abilities/absorb.dm

```diff
@@ -177,24 +177,32 @@
 	onEnd()
 		..()
 
-		var/mob/ownerMob = owner
+		var/mob/living/ownerMob = owner
 		if(owner && ownerMob && target && BOUNDS_DIST(owner, target) == 0 && devour)
 			var/datum/abilityHolder/changeling/C = devour.holder
 			if (istype(C))
 				C.addDna(target)
 			boutput(ownerMob, SPAN_NOTICE("We have absorbed [target]!"))
 			ownerMob.visible_message(SPAN_ALERT("<B>[ownerMob] sucks the fluids out of [target]!</B>"))
 			logTheThing(LOG_COMBAT, ownerMob, "absorbs [constructTarget(target,"combat")] as a changeling [log_loc(owner)].")
-
-			target.dna_to_absorb = 0
-			target.death(FALSE)
-			target.disfigured = TRUE
-			target.UpdateName()
-			target.bioHolder.AddEffect("husk")
-			target.bioHolder.mobAppearance.flavor_text = "A desiccated husk."
-
-			if (ishuman(ownerMob))
-				var/mob/living/carbon/human/H = ownerMob
+			target.full_heal()
```

Comment:
This is a *little* weird (changeling missing liver eats other person missing liver, liver appears post-absorb) but I think it's fine in the interests of cleanliness/avoiding weird edge cases. Especially considering that changelings can full heal themselves on-demand anyways.

Reply:
Yeah I considered trying to make it copy damage from the old body but that feels like a maze of edge cases and this is cleaner.